docs/dependencies.rst

```diff
@@ -0,0 +1,134 @@
+System Dependency Reference
+===========================
+
+This provides an authoritative listing of Girder's external system dependencies, for development reference.
+
+Server Install
+--------------
+Installation of Girder's server has the following system dependencies:
+
+* `Python <https://www.python.org>`_ v2.7 or v3.5+
+
+  * This is necessary to run most of the server software.
+
+  .. warning:: Some Girder plugins do not support Python v3 at this time due to third party library dependencies.
+               Namely, the ``hdfs_assetstore`` plugin and the ``metadata_extractor`` plugin will only be available in a
+               Python v2.7 environment.
+
+  .. note:: Girder performs continuous integration testing using Python v2.7 and Python v3.5. Girder *should* work on
+            newer Python v3.6+ versions as well, but that support is not verified by Girder's auftomated testing at this
+            time, so use at your own risk.
+
+* `pip <https://pip.pypa.io/>`_ v8.1+
+
+  * This is necessary to install Python packages.
+
+  * v8.1 is required to install packages via `Python wheels <https://pythonwheels.com/>`_, which significantly reduce
+    system dependencies on `manylinux <https://github.com/pypa/manylinux>`_-compatible distributions (which includes
+    almost everything except Alpine).
+
+* `setuptools <https://setuptools.readthedocs.io/>`_ v21+
+
+  * This is necessary to install Python packages.
+
+  * v21+ is required to parse Girder's ``setup.py`` file, but the latest version can and should be installed via pip.
+
+* A Python virtual environment (optional)
+
+  * This is necessary to prevent conflicts with Girder's Python dependencies and allow installation as a non-root user.
+
+  * Virtual environments are managed by the `virtualenv package <https://virtualenv.pypa.io/>`_ in Python v2.x and by
+    the built-in `venv module <https://docs.python.org/3/library/venv.html>`_ in Python v3.5+.
+
+    .. note:: Developers needing to work on several virtual environments should consider using other packages that help
```

Comment:
How necessary is this section?

Reply:
I didn't add it, and I don't feel strongly about it either way. I'd be fine with removing it in a follow-up PR.